daemons/fenced/pacemaker-fenced.c

```diff
@@ -619,16 +619,15 @@ static void
 update_stonith_watchdog_timeout_ms(xmlNode *cib)
 {
     xmlNode *stonith_enabled_xml = NULL;
-    const char *stonith_enabled_s = NULL;
+    bool stonith_enabled;
+    int rc;
     long timeout_ms = 0;
 
     stonith_enabled_xml = get_xpath_object("//nvpair[@name='stonith-enabled']",
                                            cib, LOG_NEVER);
-    if (stonith_enabled_xml) {
-        stonith_enabled_s = crm_element_value(stonith_enabled_xml, XML_NVPAIR_ATTR_VALUE);
-    }
+    rc = pcmk__xe_bool_attr(stonith_enabled_xml, XML_NVPAIR_ATTR_VALUE, &stonith_enabled);
 
-    if (stonith_enabled_s == NULL || crm_is_true(stonith_enabled_s)) {
+    if (rc == EINVAL || rc == ENODATA || (rc == pcmk_rc_ok && stonith_enabled)) {
```

Comment:
seeing it in use, i'm thinking xml==NULL makes more sense returning ENODATA rather than EINVAL

Reply:
Fixed.